Cargo.toml

```diff
@@ -17,6 +17,7 @@ edition = "2018"
 cfg-if = "1.0"
 libc = "0.2.62"
 parking_lot = ">= 0.11, < 0.13"
+memoffset = "0.6.5"
```

Comment:
Should we add a note here that it's to support safe offset calculations on versions predating `addr_of`? I imagine in the future we might consider removing again.

Reply:
I think we are using it compute offsets in all cases since even when `addr_of` is available, using `memoffset` enables us to avoid writing out its usage ourselves.

Personally, I think as it takes care of what is available where and how to use it, `memoffset` does carry its weight as a dependency.

Reply:
Agree with this, I was thinking more that after a future MSRV bump that may change as `addr_of` would be trivial for us to implement safely. I'll worry about that in the future, perhaps 😄